### PR TITLE
Add example for enabling rewrite target allocator feature

### PR DESCRIPTION
**Pull request by stuart23**
Enabling the rewrite target allocator feature is a common need when deploying this chart. A specific example in the documentation should make it easier for newcomers (like myself) to deploy the operator and scrape serviceMonitor endpoints.

### CODE REVIEW

File: charts/opentelemetry-operator/README.md [hunk 1]
```diff
@@ -49,10 +49,16 @@ $ helm install --namespace opentelemetry-operator-system \
...
 ```console
-$ helm install  --set admissionWebhooks.certManager.enabled=false --set admissionWebhooks.certManager.autoGenerateCert=true \
+$ helm install --set admissionWebhooks.certManager.enabled=false --set admissionWebhooks.certManager.autoGenerateCert=true \
   opentelemetry-operator open-telemetry/opentelemetry-operator
 ```
 
+Additional featureGates can also be enabled through the Helm installation. For enabling the [rewrite target allocator](https://github.com/open-telemetry/opentelemetry-operator#target-allocator-config-rewriting) feature, add a `--set` flag as follows.
+
+```console
+$ helm install --set manager.featureGates=operator.collector.rewritetargetallocator opentelemetry-operator open-telemetry/opentelemetry-operator
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/README.md:59]:
```suggestion
$ helm install opentelemetry-operator open-telemetry/opentelemetry-operator --set manager.featureGates=operator.collector.rewritetargetallocator
```

File: charts/opentelemetry-operator/README.md [hunk 2]
```diff
@@ -49,10 +49,16 @@ $ helm install --namespace opentelemetry-operator-system \
 If you wish for helm to create an automatically generated self-signed certificate, make sure to set the appropriate values when installing the chart:
 
 ```console
-$ helm install  --set admissionWebhooks.certManager.enabled=false --set admissionWebhooks.certManager.autoGenerateCert=true \
+$ helm install --set admissionWebhooks.certManager.enabled=false --set admissionWebhooks.certManager.autoGenerateCert=true \
   opentelemetry-operator open-telemetry/opentelemetry-operator
 ```
 
+Additional featureGates can also be enabled through the Helm installation. For enabling the [rewrite target allocator](https://github.com/open-telemetry/opentelemetry-operator#target-allocator-config-rewriting) feature, add a `--set` flag as follows.
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/README.md:56]: Lets move this into a new Operator Feature Gates section of the readme nested under the `## Configuration` section.

Comment (stuart23) [reply charts/opentelemetry-operator/README.md:56]: Sure thing

**Review by TylerHelmuth — COMMENTED** [review]
@stuart23 please bump the patch version of the chart and then run `make generate-examples CHARTS=opentelemetry-operator`

File: charts/opentelemetry-operator/README.md [hunk 3]
```diff
@@ -95,6 +95,14 @@ The following command will show all the configurable options with detailed comme
 $ helm show values open-telemetry/opentelemetry-operator
 ```
 
+Extra configuration values can be passed into the the chart individually using the `--set` flag, or in a values file using the `--values` flag as per the `helm install` [instructions](https://helm.sh/docs/helm/helm_install/).  
+
+Additional featureGates need to be enabled during the Helm installation. Enabling the [rewrite target allocator](https://github.com/open-telemetry/opentelemetry-operator#target-allocator-config-rewriting) feature gate can be done by adding a `--set` flag as follows:
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/README.md:100]:
```suggestion
Additional Operator featureGates may need to be enabled during the Helm installation. For example, enabling the [rewrite target allocator](https://github.com/open-telemetry/opentelemetry-operator#target-allocator-config-rewriting) feature gate can be done by adding a `--set` flag as follows:
```

**Comment by TylerHelmuth** [comment]
With the next release of the Operator this feature gate is being enabled by default: https://github.com/open-telemetry/opentelemetry-operator/pull/2231.  We can still add this feature gate section, but we shouldn't use the rewrite feature gate anymore.

**Pull request closed without merging**